preliminary rcs patch

Temeter opened the pull request:
Hey, I've put the two parts I've done in here, the SC-GEN-RCS-4A-V (angled single 4 way rcs), and SC-B-CM.cfg (orion reentry module).

They work fine ingame, if it looks okay to you, then I'll apply it to the others.

---

One question remains: Those EFFECT plumes use prefabs, KSP comes with two RCS plumes. It has a small white plume for monopropo, and a medium sized, yellowish plume for it's biprop rcs (stock vernour rcs):

picture: http://i.imgur.com/UIiE9Gj.png edit: better picture http://i.imgur.com/Scrtzz4.png

I think it would be cool to use both to visually reflect if a thruster uses monoprop or biprop, but that comes with a small inconsistency: The prefab size is static, so biprop plumes would _always_ be bigger than monoprop, regardless of actual thruster power (as seen in the picture).

To me that's not a problem, but it's of course your mod, so I want to ask it's fine to use both?

---
Manual conversion progress:
[x] LC-POD
[x] MRCS
[ ] MUS
[ ] MSRB
[ ] StationCore
[ ] ShipCore

Temeter committed:
Update SC-B-CM.cfg

Temeter committed:
Update SC-GEN-RCS-4F-V.cfg

shadowmage45 commented:
I think using both would be fine.

Eventually I'll be swapping most parts over to use bi-prop for RCS; only a few smaller/specialized parts will use mono (probably the probe core stuff when I get there).

Thanks for the PR, will take a look at this hopefully in the next day or two and let you know what I think :)

Temeter commented:
If you're swapping all (current) RCS to biprop, then I think it's more efficient just to make everything use one of the two prefabs for all RCS. 

Alright, just tell me which one is the best option when you've checked it (small white/medium yellow/mixed). Then I'll update the other stuff, should go rather fast.^^

Temeter commented:
Btw, I'm quite confident that I've figured the modulercsfx/effect system out. So if you decided what plume I should use, I might start right away. I'll only keep the orion sm using the old effects right now, because of the bug.

edit: Lol, don't mind the closing/opening. My mistake xD

shadowmage45 commented:
You can go ahead and update them all with the small / white plume.  If I need to adjust some for the larger effects it shouldn't be too much problem.

Sorry didn't get a chance to look at them last night, but will be making a bit of time tonight for it.  Sounds like you've got it figured out though, so I'm not too worried :)

Temeter commented:
Lol, no need to instantly jump on the issue, mate. You're making an amazing mod, so I'm happy to at least contribute a little bit, instead of just enjoying your work. ;)

I did bascially learn the system just by trying, so I was afraid I'd break some other things or references at first. But seeing how it's a simple system without to many dependencies, there isn't much to worry.

White plume is probably a bit more adequate for a realistic mod, yellow is just too big, since it's made for the vernier. I think Sarbian is also planning to do great things with 1.2's parcticle systems, so who knows whats to come? :D

Alright, going to modify some cfg's then. ~

Temeter committed:
updated literally everything :^)

literally

Temeter commented:
So, everything updated and checked ingame! :D

2 things of note:
- Parts affected by the modulercsfx bug are all upper stages, progress craft (soyuz transport thingy) and service modules (except apollo SM). Just left them as they were, we'll fix them when the game is fixed.
- While all current station rcs works, I've only touched the basic parts. Seen some unfinished things that had references to 'ModuleRCS', which might need to be 'ModuleRCSFX' if you work on them.

Also small bug report (albeit probably just unifinished); if you chose ISS solar truss to use no top or bottom adapter, then I get no texture and can see through the part.

Temeter commented:
Seems like I found an incompatability while building a saturn 5:

When using realplume, you don't get rcs plumes on the good ol' service module, while the engine effects work. Same for hab modules. I think you said something about rp removing all effects when it detects an engine plume? Might be the culprit.

Creatively thought (assuming latter assumption true): If there is an option to disable the plume removal for a certain part, and manually disable the engine plume with an sstu-own cfg...

shadowmage45 commented:
Yeah, anytime there is a plume on a part RealPlume deletes the entire EFFECTS block.

Might need a fix for that from RealPlume...

My current workaround is to redefine the needed EFFECTS through a FINAL patch after RealPlume has deleted the original values.

Temeter commented:
Alright, good to know you got it. I'd offer help, but I don't think I'm understanding how these patches are loaded.^^

Temeter committed:
rcs parts, ship+landercore (not upper stages) upgraded to

Temeter commented:
@shadowmage45 Since you've talked about it, I've also updated most aprts to hypergolic!

Namely the RCS thrusters, shipcore (minus ICS/ISUS upper stages who don't have columecontainer), and landercore (with hypergol volumecontainer).

I've found an 280 ISP stat in the orion sm module or so, and I've also applied it to all of those crafts rcs, since this is more realistic to actual aerozine50+nto. The apollo service module rcs was even 290 ISP. Is that fine? Otherwise I'll just revert it back to 260.

Btw: Only error after fixing some nullref spam was this on the landers:
`ERROR: Enabled was called for an already enabled model: AscentTanks for part: SSTU-LC2-POD (Part)`

Doesn't seem to do anything (checked parts ingame until they worked), and i've adjusted the volumecontainer using the original (and station parts) styles. Not sure what's up with that.

shadowmage45 commented:
The changes all sound fine, but am holding off on merging this one for a bit until I can work out the best way to handle the RealPlumes incompatibilities.  Probably won't make it into the next release, but should have it cleaned up for the following one.

shadowmage45 commented:
Will be taking a look at this PR over the weekend; with the fixes for the RCS module fields most of the outstanding problems should be fixed.

The one I'm still unsure on is realplumes -- not quite sure how to handle the RP setup for parts that have both engines and RCS on them without realplumes deleting the effects.  My current solution is to re-setup the RCS effects in a FINAL block, but that is not a very good option for this many parts and the large number of effects on some of them.  Would close to double the amount of config work needed.

Temeter commented:
Yeah, the RP integration does sound a bit troubling. Did you check up with felger or sarbian? Maybe they can integrate something for exceptions (depending on this being a smokescreen or realplume issue), since this is a quite obvious problem that will most likely also pop up with other mods using the new RCS system together with engine parts. SSTU's parts are only the most obvious edge case.

I'll make sure to update the remanining soyuz and orion SM modules today anyway. Hopefully that stuff is salvagable.

Temeter committed:
multiple rcs update

add soyuz SM and orion SM

Temeter committed:
orion sm correction

Temeter commented:
Alright, I've updated the two other SM's. Only untouched parts are the two upper stages, because they first need to be converted to the volume container (including hypergol), which is more complex as well as a balancing act.

One more issue (because it would be too easy otherwise): The Orion SM uses the same thrusterTransformName ('SC-B-SM-RCSThrustTransform2') for RCS number 2 and RCS number 3. This wasn't a problem for the old system. However, from what I can tell, the new plume effects want their own EFFECT **and** own thrusterTransformName for each RCS system. Had to work hours to figure that out at first, but I don't think there is any alternative.

So if you setup 2 effects with the same thrusterTransformName name, only 1 effect of them will work, which is currently the case for the orion SM (only the 4 forward pointing dual-rcs around the engine don't have an effect). All do work gameplay-wise, though.

shadowmage45 commented:
I posted up an issue on the RealPlumes repo, as that is where the EFFECTS deletion conflicts come into play.

Will take a look at the SC-B(C)/Orion Service Module.  Thought I had three separate sets of transforms, but perhaps I messed up some naming on some of them.

Temeter commented:
If there is an issue, then it's probably deeper than the cfg. Alreading tried setting it to 3, didn't work.

Temeter committed:
Merge remote-tracking branch 'refs/remotes/shadowmage45/master' into Temeter-preliminary-rcs-patch

shadowmage45 commented:
Sorry it has taken so long for me to get this merged in.  

Mostly I'm waiting on fixes on the RealPlume end of things.  As I support RealPlumes by default, I cannot merge this as-is with the current way that RealPlumes deletes all EFFECTs nodes (rcs would be broken on anything that has engines).

I did open up a bug report on RealPlumes  ( https://github.com/KSP-RO/RealPlume/issues/24 ), but RP has not seen any updating in quite awhile and I'm unsure as to when it will be fixed.

Temeter commented:
Haha, don't be sorry. Assumed it, I imagine the people behind realplume might be busy with RO or so.

We'll see how it goes. I'll be patient ~

shadowmage45 commented:
So, almost a year later, the bug in RealPlumes has finally been fixed.

Will be taking a further look at this PR over the next weeks to clean it up and update it for the current part configs / setups.

Temeter commented:
Haha, sometimes the good things take longer. Hope it's good for something!

Jimbodiah commented:
Is this still active?

shadowmage45 commented:
Kind of... I've left it here as reference to remind me that the ModuleRCSFX is a thing now.

Really need to dig into it to see what all it breaks (as undboutably it is going to break something).  That is really why this has been on hold for so long -- I don't have much time to devote to testing stuff that isn't already broken.

If you wanted to give it a test and see what all it breaks / doesn't break, we can use that to determine if work should be continued on this patch or not.

Jimbodiah commented:
I didn't actually read what it was about, just that it's been on the list so long I thought you forgot to remove it or something :)

Always willing to test stuff...

shadowmage45 commented:
Have manually implemented these changes, as they were needed for run-time ModuleRCS manipulation in ModularPart.
